minesweeper.py

```diff
@@ -293,7 +295,8 @@ def save_json_file():
     json_dict["height_slider"] = int(height_slider.scale.get())
     json_dict["width_slider"] = int(width_slider.scale.get())
     json_dict["difficulty_slider"] = int(difficulty_slider.scale.get())
-    with open(image_dir / "game_data.json", "w") as file:
+    pathlib.Path(config_dir).mkdir(exist_ok=True, parents=True)
```

Comment:
Because `config_dir` is alraedy a pathlib.Path object, there's no need to convert it:

```suggestion
    config_dir.mkdir(exist_ok=True, parents=True)
```